tests/stdlib/trstgen.nim

```diff
@@ -7,9 +7,34 @@ outputsub: ""
 import ../../lib/packages/docutils/rstgen
 import ../../lib/packages/docutils/rst
 import unittest, strutils, strtabs
-
-proc toHtml(input: string): string =
-  rstToHtml(input, {roSupportMarkdown}, defaultConfig())
+import std/private/miscdollars
+
+proc toHtml(input: string,
+            rstOptions: RstParseOptions = {roSupportMarkdown},
+            error: ref string = nil,
+            warnings: ref seq[string] = nil): string =
+  ## If `error` is nil then no errors should be generated.
+  ## The same goes for `warning`.
+  proc testMsgHandler(filename: string, line, col: int, msgkind: MsgKind,
+                      arg: string) =
+    let mc = msgkind.whichMsgClass
+    let a = $msgkind % arg
+    var message: string
+    toLocation(message, filename, line, col)
+    message.add " $1: $2" % [$mc, a]
+    if mc == mcError:
+      doAssert error != nil, "unexpected RST error '" & message & "'"
+      error[] = message
+      # we check only first error because subsequent ones may be meaningless
+      raise newException(EParseError, message)
```

Comment:
why raise only to catch later? is this to stop on 1st error? can't we instead:
* either keep track of number of errors seen here and not overwrite `error[]` after it's been written to once
* or add an option roStopOnFirstError

Reply:
yes, just to stop on 1st error.

We can do all of that but there is little sense. In the 1st commit all errors were checked but I saw that they add little value to 1st error. Check for number of errors is better but that would be difficult to develop: if you broke tests around this number then you don't know immediately what those errors were :-)

Reply:
what i meant was keeping track of number of errors so you only set `error[]` for just the 1st emitted error, and then ignoring subsequent ones. but ok, it's a minor detail.